internal/controller/authenticator.go

```diff
@@ -181,11 +194,27 @@ func SetUpOAuthConnection(
 	}
 	tokenSource = GetTokenSource(ctx, oauthConfig, tlsConfig)
 	creds := credentials.NewTLS(tlsConfig)
+
+	proxyDialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		proxyFunc := http.ProxyFromEnvironment
+		proxyURL, _ := proxyFunc(&http.Request{URL: &url.URL{Host: addr}})
+		if proxyURL == nil { // No proxy configured
+			return net.Dial("tcp", addr)
+		}
+		conn, err := proxy.Dial(ctx, "tcp", addr)
```

Comment:
Why are you ignoring `proxyURL` and connect to the CloudSecure address anyway?
Have you really tested this?
```suggestion
		conn, err := proxy.Dial(ctx, "tcp", proxyURL.Host)
```